tests/test_unit.py

```diff
@@ -5148,61 +4990,33 @@ def getNameTester(s, l, t):
             self.assertEqual(len(stmt[1, ...]('tests').parseString('test test').tests), 2, "multipled(3) failure with setResultsName")
             self.assertEqual(len(stmt[2, ...]('tests').parseString('test test').tests), 2, "multipled(3) failure with setResultsName")
 
-def makeTestSuite():
-    import inspect
-    suite = TestSuite()
-    suite.addTest(PyparsingTestInit())
-
-    test_case_classes = ParseTestCase.__subclasses__()
-    # put classes in order as they are listed in the source code
-    test_case_classes.sort(key=lambda cls: inspect.getsourcelines(cls)[1])
 
-    test_case_classes.remove(PyparsingTestInit)
-    # test_case_classes.remove(ParseASMLTest)
-    if IRON_PYTHON_ENV:
-        test_case_classes.remove(OriginalTextForTest)
-
-    suite.addTests(T() for T in test_case_classes)
-
-    if TEST_USING_PACKRAT:
-        # retest using packrat parsing (disable those tests that aren't compatible)
-        suite.addTest(EnablePackratParsing())
+class PickleTest_Greeting():
+    def __init__(self, toks):
+        self.salutation = toks[0]
+        self.greetee = toks[1]
 
-        unpackrattables = [PyparsingTestInit, EnablePackratParsing, RepeaterTest,]
+    def __repr__(self):
+        return "%s: {%s}" % (self.__class__.__name__,
+                             ', '.join('%r: %r' % (k, getattr(self, k)) for k in sorted(self.__dict__)))
 
-        # add tests to test suite a second time, to run with packrat parsing
-        # (leaving out those that we know wont work with packrat)
-        packratTests = [t.__class__() for t in suite._tests
-                            if t.__class__ not in unpackrattables]
-        suite.addTests(packratTests)
 
-    return suite
+class Test3_EnablePackratParsing(TestCase):
+    def runTest(self):
+        ParserElement.enablePackrat()
 
-def makeTestSuiteTemp(classes):
-    suite = TestSuite()
-    suite.addTest(PyparsingTestInit())
-    suite.addTests(cls() for cls in classes)
-    return suite
+        # SAVE A NEW SUITE CONTEXT
+        Test2_WithoutPackrat.suite_context = ppt.reset_pyparsing_context()
+        Test2_WithoutPackrat.suite_context.save()
 
-# runnable from setup.py using "python setup.py test -s unitTests.suite"
-suite = makeTestSuite()
+Test4_WithPackrat = type(
+    "Test4_WithPackrat",
+    (Test2_WithoutPackrat,),
+    {}
+)
```

Comment:
This declares a new class, that inherits from `(Test2_WithoutPackrat,)` and adds no new attributes.  Now we have two classes that do the same thing, but with packrat mode being different.